lazy_dataset/core.py

```diff
@@ -147,6 +150,98 @@ def concatenate(*datasets):
     return ConcatenateDataset(*datasets)
 
 
+def _zip(*datasets):
+    """
+    Create a new `Dataset` zipping all passed datasets.
+
+    Example:
+        >>> import lazy_dataset
+        >>> ds1 = lazy_dataset.new({'1': 1, '2': 2, '3': 3, '4': 4})
+        >>> ds2 = lazy_dataset.new({'1': 5, '2': 6, '3': 7, '4': 8})
+        >>> lazy_dataset.zip(ds1, ds2)
+            DictDataset(len=4)
+          MapDataset(_pickle.loads)
+            DictDataset(len=4)
+          MapDataset(_pickle.loads)
+        ZipDataset()
+
+        >>> lazy_dataset.zip((ds1, ds2))
+            DictDataset(len=4)
+          MapDataset(_pickle.loads)
+            DictDataset(len=4)
+          MapDataset(_pickle.loads)
+        ZipDataset()
+
+        >>> list(lazy_dataset.zip(ds1, ds2))
+        [(1, 5), (2, 6), (3, 7), (4, 8)]
+
+    Args:
+        datasets: List of datasets. Can be either a list of datasets
+            (`zip_((ds1, ds2, ...))`) or multiple datasets
+            (`zip_(ds1, ds2, ...)`)
+
+    Returns:
+        zip of all input datasets
+
+    """
+    if len(datasets) == 0:
+        raise ValueError('Need at least one dataset to concatenate!')
+    if len(datasets) == 1 and isinstance(datasets[0], (tuple, list)):
+        datasets, = datasets
+    if not all([isinstance(dataset, Dataset) for dataset in datasets]):
+        raise TypeError(
+            f'All input arguments must be datasets! {Dataset} ' + ' '.join(
+                str(type(d)) for d in datasets) + '|' + ' '.join(
+                str(isinstance(d, Dataset)) for d in datasets))
```

Comment:
`str(type(d)) for d in datasets)` will yield something like `'<generator object <genexpr> at 0x7f6ea87e7308>'`.
Could you add the `[` and `]`?

Reply:
This code is fine, I overlooked one closing bracket